### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by jusabatier**
Update Native Libs Doc : 
- Install from debian package
- Compile GDAL (with ecw support)

Update Optimisation Doc : 
- Add ms-fonts install
- Add non-free deposit for jai install

Correct Cluster Doc

### CODE REVIEW

File: doc/gs_cluster.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Those commands will install TC8 and optimisation for GS like TurboJPEG, Marlin R
 Then open the */etc/default/tomcat8* file and set **JAVA_OPTS** to : 
 
 ```
--Djava.awt.headless=true -XX:+UseConcMarkSweepGC -Xms2G -Xmx2G -XX:PermSize=256m -DGEOSERVER_DATA_DIR=/opt/geoserver_datadir -Dfile.encoding=UTF8 -Djavax.servlet.request.encoding=UTF-8 -Djavax.servlet.response.encoding=UTF-8 -server -XX:+UseParNewGC -XX:ParallelGCThreads=2 -XX:SoftRefLRUPolicyMSPerMB=36000 -XX:NewRatio=2 -XX:+AggressiveOpts -Djava.library.path=/usr/lib/jni:/opt/libjpeg-turbo/lib64
```
Comment (fvanderbiest) [inline doc/gs_cluster.md]: A comment regarding -XX:+UseConcMarkSweepGC removal ?

File: doc/optimizations.md [hunk 2]
```diff
@@ -23,12 +30,12 @@ This is usually done by symlinking them from their original location (something
...
-sudo ln -s /usr/share/java/mlibwrapper_jai.jar .
-sudo ln -s /usr/share/java/jai_imageio.jar .
-sudo ln -s /usr/share/java/clibwrapper_jiio.jar .
-sudo service tomcat-geoserver0 restart
+ln -s /usr/share/java/jai_core.jar .
+ln -s /usr/share/java/jai_codec.jar .
+ln -s /usr/share/java/mlibwrapper_jai.jar .
+ln -s /usr/share/java/jai_imageio.jar .
+ln -s /usr/share/java/clibwrapper_jiio.jar .
+service tomcat-geoserver0 restart
```
Comment (fvanderbiest) [inline doc/optimizations.md]: why did you remove sudo in some places and kept it at other places ?

File: doc/optimizations.md [hunk 3]
```diff
@@ -102,6 +109,22 @@ If you have followed this guide, your geoserver probably also uses our recommend
 If not, you should create a custom ```controlflow.properties``` file in your geoserver "data dir".  
 Please refer to the [control-flow module documentation](http://docs.geoserver.org/latest/en/user/extensions/controlflow/index.html) for the syntax.
 
+### More fonts
```
Comment (fvanderbiest) [inline doc/optimizations.md]: We should probably remove the same wording from https://github.com/georchestra/georchestra/blob/15.06/doc/post-deploy_config.md#fonts 

And clearly say that this should be performed on the machine hosting the geoserver.

File: doc/setup/native_libs.md [hunk 4]
```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
...
 
-To quickly install the GDAL Java binding library on the server, download and extract the library and its data. To do so, you can use 2 repositories providing binary packages for given distribution:
+For example, in order to enable the support of ECW format, we need to install it. We will explain how to install the version 2-3.3 of the ECW SDK, but this isn't the newer version and you can choose one more recent if you need but remember there are limits for the uses of this SDK with no licence.
 
- * GeoSolutions http://demo.geo-solutions.it/share/github/imageio-ext/releases/1.1.X/1.1.7/native/gdal/ which provides a package with ECW support (which can cause some licensing issues, and some of the packages are outdated, since the distributions evolved and updated their Glibc version, see #409) ;
- * geOrchestra-provided packages http://sdi.georchestra.org/~pmauduit/gdalogr-java-bindings/ which provides "vanilla" GDAL packages as well as a specific one (mifmid-patched) which allows the use of MIF/MID format across GDAL/OGR via GeoTools (see #409)
+First search `libecwj2-3.3-2006-09-06.zip` on Google in order to found the SDK. (no link because it always change)
```
Comment (fvanderbiest) [inline doc/setup/native_libs.md]: /Google/any search engine/

File: doc/setup/native_libs.md [hunk 5]
```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
...
 
+After you get this zip simply, we'll need to apply some patch to correct the jp2K support and som other bugs and then compile the sources and install it : 
+```
+unzip libecwj2-3.3-2006-09-06.zip
+rm libecwj2-3.3-2006-09-06.zip
+wget http://trac.osgeo.org/gdal/raw-attachment/ticket/3162/libecwj2-3.3-msvc90-fixes.patch
+patch -p0 < libecwj2-3.3-msvc90-fixes.patch
+wget http://osgeo-org.1560.x6.nabble.com/attachment/5001530/0/libecwj2-3.3-wcharfix.patch
+wget http://trac.osgeo.org/gdal/raw-attachment/ticket/3366/libecwj2-3.3-NCSPhysicalMemorySize-Linux.patch
```
Comment (fvanderbiest) [inline doc/setup/native_libs.md]: I suggest you create a gist with those files, and mention the gist URLs rather than those

File: doc/setup/native_libs.md [hunk 6]
```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
...
+
+The advantage to compile GDAL is to allow more formats to be supported (ecw, jp2K, ...), you can choose the ones which suits your needs.
+
+In order to compile we'll need some compiler packages :
+```
+apt-get install build-essential
+```
 
-Some more work is needed for installing the GDAL Java binding library, as there is still no deb package for it (note that packages exist for ruby and perl bindings, hopefully the Java's one will be released soon - see a recent proposal http://ftp-master.debian.org/new/gdal_1.10.0-0%7Eexp3.html).
+#### Enable ECW support
```
Comment (fvanderbiest) [inline doc/setup/native_libs.md]: Mention that this is an optional step, and that carefully crafted GeoTiff pyramids are as fast as ECW.

**Comment by fvanderbiest** [comment]
Thanks a lot for helping us improve the documentation Julien :-)

File: doc/gs_cluster.md (hunk 1 above)
Comment (jusabatier) [inline doc/gs_cluster.md]: I removed this option because it has been removed from the GS part of the tomcat setup doc.

File: doc/optimizations.md (hunk 2 above)
Comment (jusabatier) [inline doc/optimizations.md]: I miss one sudo in my correction. I remove them because 'sudo' is used in Ubuntu, not in debian by default.

File: doc/setup/native_libs.md (hunk 6 above)
Comment (jusabatier) [inline doc/setup/native_libs.md]: done

File: doc/setup/native_libs.md (hunk 4 above)
Comment (jusabatier) [inline doc/setup/native_libs.md]: done

File: doc/setup/native_libs.md (hunk 5 above)
Comment (jusabatier) [inline doc/setup/native_libs.md]: done

**Comment by fvanderbiest** [comment]
Would be nice if we could get a review from @pmauduit before we merge.

File: doc/setup/native_libs.md [hunk 7]
```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
...
+apt-get install build-essential
 ```
 
+#### Enable ECW support (optional)
+
+Note : a carefully crafted GeoTiff pyramid is as fast as ECW.
+
+For example, in order to enable the support of ECW format, we need to install it. We will explain how to install the version 2-3.3 of the ECW SDK, but this isn't the newer version and you can choose one more recent if you need but remember there are limits for the uses of this SDK with no licence.
+
+First search `libecwj2-3.3-2006-09-06.zip` on any search engine in order to found the SDK. (no link because it always change)
```
Comment (pmauduit) [inline doc/setup/native_libs.md]: *in order to find

File: doc/setup/native_libs.md [hunk 8]
```diff
@@ -14,47 +14,116 @@ The key element for calling the GDAL native library from mapfishapp is the **ima
...
+
+After you get this zip simply, we'll need to apply some patch to correct the jp2K support and som other bugs and then compile the sources and install it :
```
Comment (pmauduit) [inline doc/setup/native_libs.md]: *some other bugs

**Comment by pmauduit** [comment]
Looks good, thanks !

File: doc/gs_cluster.md (hunk 1 above)
Comment (fvanderbiest) [inline doc/gs_cluster.md]: GeoServer doc recommends -XX:+UseParallelGC, see http://docs.geoserver.org/stable/en/user/production/container.html but I guess it's outdated

GeoSolutions recommends -XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:ParallelGCThreads=4 as per http://geoserver.geo-solutions.it/edu/en/adv_gsconfig/gsproduction.html

Reading http://stackoverflow.com/questions/2101518/difference-between-xxuseparallelgc-and-xxuseparnewgc makes me think that GeoSolutions is right:
```
Apply -XX:+UseParNewGC & -XX:+UseConcMarkSweepGC when you require parallel collection method over YOUNG generation AND require CMS method as your collection over OLD generation memory
```

I'm more concerned by the validity of the other options mentioned here (-XX:SoftRefLRUPolicyMSPerMB=36000 -XX:NewRatio=2 -XX:+AggressiveOpts)

Comment (jusabatier) [inline doc/gs_cluster.md]: So we should use : 
```
-XX:+UseParNewGC -XX:+UseConcMarkSweepGC -XX:ParallelGCThreads=4
```
in the georchestra doc ?

I think we can remove `-XX:NewRatio=2` because it's the default value of the JVM : http://www.oracle.com/technetwork/java/javase/tech/vmoptions-jsp-140102.html

Same for `-XX:+AggressiveOpts` which seems to be default activated on version upper 5.0 update 6

Can you confirm it ?

Comment (fvanderbiest) [inline doc/gs_cluster.md]: Go ahead

**Comment by fvanderbiest** [comment]
Thanks again Julien !
